warehouse/query-core/src/main/java/datawave/query/iterator/logic/AndIterator.java

```diff
@@ -178,12 +236,8 @@ public T move(T minimum) {
     
     public Collection<NestedIterator<T>> leaves() {
         LinkedList<NestedIterator<T>> leaves = new LinkedList<>();
-        for (NestedIterator<T> itr : includes) {
-            leaves.addAll(itr.leaves());
-        }
-        for (NestedIterator<T> itr : excludes) {
-            leaves.addAll(itr.leaves());
-        }
+        // treat this node as a leaf as it's seek
+        leaves.add(this);
```

Comment:
this will mean that the call in SeekableNestedIterator will bring back this node in its leaf set.

Reply:
Since the AndIterator is a SeekableIterator, the seek method of SeekableNestedIterator has and always will just call seek directly on the AndIterator.  It won't attempt to get the AndIterator's leaves because the AndIterator itself is seekable.

Reply:
um...is that wrong?

Reply:
To me it seems consistent with what the existing code is doing.  @FineAndDandy ?

Reply:
any leaves that are underneath this AND will be seeked again when it loops over the leaves

Reply:
I think this is what I need, because I need to code to pass through the seek method on the AndIterator.